hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsInputStream.java

```diff
@@ -564,11 +669,25 @@ int readRemote(long position, byte[] b, int offset, int length, TracingContext t
     } catch (AzureBlobFileSystemException ex) {
       if (ex instanceof AbfsRestOperationException) {
         AbfsRestOperationException ere = (AbfsRestOperationException) ex;
+        abfsHttpOperation = ((AbfsRestOperationException) ex).getAbfsHttpOperation();
         if (ere.getStatusCode() == HttpURLConnection.HTTP_NOT_FOUND) {
           throw new FileNotFoundException(ere.getMessage());
         }
+        /*
+         * Status 416 is sent when read range is out of contentLength range.
+         * This would happen only in the case if contentLength is not known before
+         * opening the inputStream.
+         */
+        if (ere.getStatusCode() == READ_PATH_REQUEST_NOT_SATISFIABLE
+            && !fileStatusInformationPresent.get()) {
+          return -1;
+        }
       }
       throw new IOException(ex);
+    } finally {
+      if (!fileStatusInformationPresent.get() && abfsHttpOperation != null) {
+        initPathPropertiesFromReadPathResponseHeader(abfsHttpOperation);
```

Comment:
method name can be shortened

Reply:
sure, have refactored to `initPropertiesFromReadResponseHeader`.